internal/docs/design.md

```diff
@@ -374,6 +375,100 @@ b, err := blob.NewBucket(d, blob.FeatureUnicodeNames)
 Design discussions regarding enforcing portability are ongoing; we welcome input
 on the [mailing list](https://groups.google.com/forum/#!forum/go-cloud).
 
+### Strings
+
+Providers often differ on what characters they accept in particular strings
+(e.g., blob names, metadata keys, etc.). A couple of specific examples:
+
+*   Azure Blob only
+    [accepts C# identifiers](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-properties-metadata)
+    as metadata keys.
+*   S3 drops double slashes in blob names (e.g., `foo//bar` will end up being
+    saved as `foo/bar`).
+
+These differences lead to a loss of portability and predictability for users.
+
+We considered restricting Go CDK's APIs to strings that all providers support.
+For example, we asserting that Go CDK's blob only supports ASCII plus `/` for
+blob names (and no `//`!). However, such a rule would mean that we couldn't
+handle existing strings that were created through some mechanism other than
+through Go CDK APIs. For example, an existing blob in S3 with a unicode name
```

Comment:
I don't know if this example sells this approach: wouldn't this still be the case after this design? An existing blob in S3 with a Unicode name still would not be visible or writable.

Reply:
Yeah, it's a prose organization issue. This section buries the lede. Most other sections in this doc will present the guidance first, and then at the very end conclude with the rationale.